### PR TITLE
basic api functionality for user (get post put delete)

### LINKED ISSUE #17: Add API Routes for Users

### PR DESCRIPTION
**Pull request by wjrhee**
closes #17

### CODE REVIEW

File: server/app/routes/user.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+var router = require('express').Router();
+var User = require('/../../db').model('User');
+
+
+router.get('/users/:id', function(req, res, next){
+  return User.findById(req.params.id)
```
Comment (Abazhenov) [inline server/app/routes/user.js]: I think these should be in an /api/ route

File: server/app/routes/user.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    })
+    .catch(function(e){
+      res.sendStatus(500);
+    })
+})
+
+router.get('/users', function(req, res, next){
+  return User.findAll()
+    .then(function(users){
+      // what fields should we send back???
```
Comment (Abazhenov) [inline server/app/routes/user.js]: Lets do email, firstname, and lastname for now

**Comment by Abazhenov** [comment]
commented on some potential improvement, also think you should add comments to put / post / delete that they are totally unsecure so we don't forget to come back and fix them.

File: server/app/routes/user.js (hunk 1 above)
Comment (connieelee) [inline server/app/routes/user.js]: @Abazhenov they already are in an /api route (check out app/index.js), but I would consider adding a '/users' router in routers/user.js so you don't have to put "/users/" in front of every route in users.js

**Comment by connieelee** [comment]
Consider refactoring your user api routes into a `router.use('/user', require('./user'));` in `app/routes/index.js`

File: server/app/routes/user.js (hunk 1 above)
Comment (Abazhenov) [inline server/app/routes/user.js]: I think that you should rename the routes folder apiRoutes so that we can have other routes if we need them.

**Pull request merged**